### PR TITLE
[VarDumper] Remove leading 0 in microseconds of date caster

### PR DESCRIPTION
**Pull request by maidmaid**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | /
| License       | MIT
| Doc PR        | /

It looks more like a [engineering notation](https://en.wikipedia.org/wiki/Engineering_notation). We can now better discern **milli**seconds and **micro**seconds.

```php
// before -> after
1.000000 -> 1.0
1.100000 -> 1.100
1.120000 -> 1.120
1.123000 -> 1.123
1.123400 -> 1.123400
1.123450 -> 1.123450
1.123456 -> 1.123456
```

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php [hunk 1]
```diff
@@ -32,13 +32,36 @@ public static function castDateTime(\DateTimeInterface $d, array $a, Stub $stub,
...
+     * 1.123450 -> 1.123450
+     * 1.123456 -> 1.123456
+     * </code>
+     *
+     * @param $s  int|string seconds
+     * @param $us int|string microseconds
+     *
+     * @return string formatted seconds
+     */
+    private static function formatSec($s, $us)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: private methods should be added at the end of the file

Comment (maidmaid) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: Moved

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php [hunk 2]
```diff
@@ -79,4 +79,27 @@ public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stu
...
+    {
+        return sprintf('%02d.%s', $s, 0 === ($len = strlen($t = rtrim($us, '0'))) ? '0' : ($len <= 3 ? str_pad($t, 3, 0) : $us));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: `str_pad($t, 3, '0')`

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php [hunk 3]
```diff
@@ -79,4 +79,27 @@ public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stu
 
         return $filter & Caster::EXCLUDE_VERBOSE ? $z : $z + $a;
     }
+
+    /**
+     * Formats seconds.
+     *
+     * <code>
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: no need for this code sample for a private method, tests are here to show examples

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php [hunk 4]
```diff
@@ -79,4 +79,27 @@ public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stu
...
+     * 1.000000 -> 1.0
+     * 1.100000 -> 1.100
+     * 1.120000 -> 1.120
+     * 1.123000 -> 1.123
+     * 1.123400 -> 1.123400
+     * 1.123450 -> 1.123450
+     * 1.123456 -> 1.123456
+     * </code>
+     *
+     * @param $s  int|string seconds
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: type is reveresed, should be `string $s` (no need for "int" imho)

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php (hunk 2 above)
Comment (maidmaid) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: Fixed

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php (hunk 3 above)
Comment (maidmaid) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: Removed

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php (hunk 4 above)
Comment (maidmaid) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: Fixed

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php [hunk 5]
```diff
@@ -79,4 +79,17 @@ public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stu
...
+
+    /**
+     * Formats seconds.
+     *
+     * @param string $s  seconds
+     * @param string $us microseconds
+     *
+     * @return string formatted seconds
+     */
+    private static function formatSec($s, $us)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: sorry, last comment: formatSeconds (then, I don't see much value in the docblock, so if you want you can remove it :) )

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php [hunk 6]
```diff
@@ -79,4 +79,9 @@ public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stu
 
         return $filter & Caster::EXCLUDE_VERBOSE ? $z : $z + $a;
     }
+
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: trailing space :)

Comment (maidmaid) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: :skull:

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @maidmaid.

**Pull request merged**